rules/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Rector\Transform\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PHPStan\Type\ObjectType;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\PhpVersion;
+use Rector\PhpAttribute\Printer\PhpAttributeGroupFactory;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://github.com/php/php-src/pull/7571
+ *
+ * @see \Rector\Tests\Transform\Rector\Class_\AddAllowDynamicPropertiesAttributeRector\AddAllowDynamicPropertiesAttributeRectorTest
+ */
+class AddAllowDynamicPropertiesAttributeRector extends AbstractRector
+{
+    /**
+     * @var string
+     */
+    private const ATTRIBUTE = 'AllowDynamicProperties';
+
+    public function __construct(
+        private PhpAttributeGroupFactory $phpAttributeGroupFactory,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add the `AllowDynamicProperties` attribute to all classes', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeObject {
+    public string $someProperty = 'hello world';
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+#[AllowDynamicProperties]
+class SomeObject {
+    public string $someProperty = 'hello world';
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $alreadyHasAttribute = false;
+        foreach ($node->attrGroups as $attrGroup) {
+            foreach ($attrGroup->attrs as $key => $attribute) {
+                if ($this->isName($attribute->name, 'AllowDynamicProperties')) {
+                    $alreadyHasAttribute = true;
+                    break 2;
+                }
+            }
+        }
```

Comment:
you can use `PhpAttributeAnalyzer` service for it:

https://github.com/rectorphp/rector-src/blob/8c31c3a6b99b32054f5b9baa5d2eeedd44c2f22a/rules/Php80/NodeAnalyzer/PhpAttributeAnalyzer.php#L20

First, you need to inject it in construct, like in : 

https://github.com/rectorphp/rector-src/blob/8c31c3a6b99b32054f5b9baa5d2eeedd44c2f22a/rules/Php80/Rector/Class_/DoctrineAnnotationClassToAttributeRector.php#L70

then, check it:

https://github.com/rectorphp/rector-src/blob/8c31c3a6b99b32054f5b9baa5d2eeedd44c2f22a/rules/Php80/Rector/Class_/DoctrineAnnotationClassToAttributeRector.php#L246

Reply:
Thank you @samsonasik - great suggestions, updated now.